pymatgen/core/structure.py

```diff
@@ -284,7 +284,7 @@ def species_and_occu(self) -> list[Composition]:
         return [site.species for site in self]
 
     @property
-    @deprecated(message="Use n_type_sp instead")
+    @deprecated(message="Use n_type_sp instead")  # DEBUG: no such property
```

Comment:
`SiteCollection` doesn't seem to have a `n_type_sp` `property`?

Reply:
yeah, looks like the deprecation message was meant to say "Use `n_elems` instead". i think we can just use `replacement=n_elems`

Reply:
Yes. However I don't quite know how to refer to a method/property of a class in such a decorator, `self.n_elems` nor `SiteCollection.n_elems` seem to work?

Reply:
Solved, swapping the order fixed this :)
```
    @property
    def n_elems(self) -> int:
        """Number of types of atoms."""
        return len(self.types_of_species)

    @property
    @deprecated(message="Use n_type_sp instead", replacement=n_elems)
    def ntypesp(self) -> int:
        """Number of types of atoms."""
        return len(self.types_of_species)
```